main-app/app/scripts/Proxy/objectConverter.js

```diff
@@ -0,0 +1,69 @@
+(function () {
+    'use strict';
+    angular.module('Tombola.BingoClient.ObjectConverter')
```

Comment:
This would be simpler if you used the converter in the layer above the base proxy - for example you know when you are logging in or out - so just use the converter in that method without the need for a if statement etc.